toml/_parser.ts

```diff
@@ -115,6 +115,15 @@ export class Scanner {
   startsWith(searchString: string) {
     return this.#source.startsWith(searchString, this.#position);
   }
+
+  match(regExp: RegExp) {
+    regExp = new RegExp(
```

Comment:
Can we give `y` flag to the original regexp instance and avoid creating new regexp each time?

Reply:
Yes we could. I just worry about accidental missing sticky flags which might introduce unintended behavior/bugs and headaches in the future.
How about we add a check just to be on the safe side?
```ts
if (!regExp.sticky) {
  throw new Error(`RegExp ${regExp} does not have a sticky 'y' flag`);
}
```

Reply:
> How about we add a check just to be on the safe side?

Sounds a good idea to me.